android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -259,43 +254,45 @@ protected void onResume() {
     KMKeyboardDownloaderActivity.addKeyboardDownloadEventListener(this);
     PackageActivity.addKeyboardDownloadEventListener(this);
 
+    CheckInstallReferrer.checkGooglePlayInstallReferrer(this, context);
+
     Intent intent = getIntent();
-    data = intent.getData();
+    Uri loadingIntentUri = intent.getData();
 
-    if (data != null) {
-      String scheme = data.getScheme().toLowerCase();
+    if (loadingIntentUri != null) {
+      String scheme = loadingIntentUri.getScheme().toLowerCase();
       switch (scheme) {
         // content:// Android DownloadManager
         // file:// Chrome downloads and Filebrowsers
         case "content":
         case "file":
-          checkStoragePermission(data);
+          checkStoragePermission(loadingIntentUri);
           break;
         case "http" :
         case "https" :
           // Might need to modify link like KMPBrowserActivity
-          String link = data.toString();
+          String link = loadingIntentUri.toString();
           if (KMPLink.isKeymanInstallLink(link)) {
-            data = KMPLink.getKeyboardDownloadLink(link);
+            loadingIntentUri = KMPLink.getKeyboardDownloadLink(link);
           }
-          downloadKMP(scheme);
+          downloadKMP(loadingIntentUri);
           break;
         case "keyman" :
           // TODO: Only accept download links from Keyman browser activities when universal links work
-          if (KMPLink.isKeymanDownloadLink(data.toString())) {
+          if (KMPLink.isKeymanDownloadLink(loadingIntentUri.toString())) {
```

Comment:
Is this still a TODO if we use referrer instead of universal links?

Reply:
I think we leave that for the bigger tidyup of the package download